src/lib/Handshake.js

```diff
@@ -87,76 +88,81 @@ const ID_BYTES = 12;
 const SESSION_BYTES = 40;
 const GLOBAL_TIMEOUT = 10;
 
+// TODO make Handshake module stateless.
 class Handshake {
   _client: Device;
+  _deviceKeyRepository: Repository<string>;
   _socket: Socket;
   _handshakeStage: HandshakeStage = 'send-nonce';
   _reject: ?Function;
-  _deviceID: string = '';
+  _deviceID: string;
   _pendingBuffers: Array<Buffer> = [];
   _useChunkingStream: boolean = true;
 
-  constructor(client: Device) {
-    this._client = client;
-    this._socket = client._socket;
+  constructor(deviceKeyRepository: Repository<string>) {
+    this._deviceKeyRepository = deviceKeyRepository;
   }
 
-  start = async (): Promise<*> => {
+  start = async (device: Device): Promise<*> => {
+    this._client = device;
```

Comment:
Yeah, maybe pass in some IDs or whatever is needed for logging.  They actually help out a lot when debugging.